compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp

```diff
@@ -462,15 +458,6 @@ static bool isFusableWithConsumer(
     }
   }
 
-  // Check if the iteration spaces of the producer and consumer are same.
-  // TODO: This is unnecessary requirement, but needed to pass tests right now
-  if (!aggressiveFusion) {
-    auto producerIterationSpace = producerLinalgOp.getStaticLoopRanges();
-    auto consumerIterationSpace = consumerLinalgOp.getStaticLoopRanges();
-    if (producerIterationSpace.size() < consumerIterationSpace.size()) {
-      return false;
-    }
-  }
```

Comment:
The removal of this check seems fine to me... At least from the benchmark results

Reply:
Hmm. It seems like in this case we won't tile the inner parallel loops of the consumer. Put the condition back and filed https://github.com/openxla/iree/issues/12664